### ISSUE TITLE
Cancelling BNLoadBinaryView with progress causes SIGSEGV

### ISSUE THREAD

**Issue by CouleeApps** [Vector35/binaryninja-api#5254]
**Version and Platform (required):**
- Binary Ninja Version: 4.1.5071-dev
- OS: macOS (also windows)
- OS Version: 14.0
- CPU Architecture: M1

**Bug Description:**
When using the python api, you can cause a SIGSEGV if you return the wrong thing from a progress function. Even though you may have returned the wrong type, that is still not an excuse to crash.

**Steps To Reproduce:**
1. Find or create a bndb
2. Run the following: `load("path/to/file.bndb", progress_func=lambda a,b:None)`
3. Observe crash

**Expected Behavior:**
I expected an exception or error message, but not a SEGV/crash.

**Additional Information:**
Reported by an Enterprise customer. Crash log: [binaryninja-2024-04-09-222541.ips.zip](https://github.com/Vector35/binaryninja-api/files/14926351/binaryninja-2024-04-09-222541.ips.zip)

**Comment by CouleeApps** [comment]
This may not be the fault of the progress function. Indeed the crashing code is actually an error message:

```
if (!view)
{
	LogError("Unable to open existing database with filename %s", view->GetFile()->GetFilename().c_str());
	return nullptr;
}
```

**Comment by CouleeApps** [comment]
Fixed in 4.1.5113+
